firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -323,13 +323,16 @@ public void handleCredentialChange() {
 
   // Watch Stream
 
-  /** Listens to the target identified by the given QueryData. */
+  /**
+   * Listens to the target identified by the given QueryData.
+   *
+   * <p>It is a no-op if the target of the given query data is already being listened to.
```

Comment:
Why do we make it a no-op to double-listen or double-unlisten from queries?  It seems like the calling code should still be doing the proper accounting (if we double-unlisten that suggests that we unlistened too early and it's a bug)...  So I'd be inclined to keep this strict, in the hopes of preventing potential bugs.

Reply:
See above.